liota/dccs/generic_mqtt.py

```diff
@@ -0,0 +1,147 @@
+        """
+        :param con: AWSMQTTDccComms Object
+        :param enclose_metadata: Include Gateway, Device and Metric names as part of payload or not
+        """
+        super(GenericMqtt, self).__init__(con)
+        self.enclose_metadata = enclose_metadata
+
+    def register(self, entity_obj):
+        log.info("Registering resource with GenericMqtt DCC {0}".format(entity_obj.name))
+        if isinstance(entity_obj, Metric):
+            return RegisteredMetric(entity_obj, self, None)
+        else:
+            return RegisteredEntity(entity_obj, self, None)
+
+    def create_relationship(self, reg_entity_parent, reg_entity_child):
+        """
+        This method creates Parent-Child relationship.  Supported relationships are:
+
+               EdgeSystem
+                   |                                      EdgeSystem
+                Device                   (or)                |
+                   |                                    RegisteredMetric
+             RegisteredMetric
+
+        However, A single EdgeSystem can have multiple child Devices and a each Device can have
+        multiple child Metrics.
+
+        :param reg_entity_parent: Registered EdgeSystem or Registered Device Object
+        :param reg_entity_child:  Registered Device or Registered Metric Object
+        :return: None
+        """
+
+        # Validating here helps in enclose_metadata
+        if not isinstance(reg_entity_parent.ref_entity, EdgeSystem) and \
+                not isinstance(reg_entity_parent.ref_entity, Device):
+            raise TypeError("reg_entity_parent should either be a Registered EdgeSystem or a Device")
+
+        if not isinstance(reg_entity_child.ref_entity, Device) and \
+                not isinstance(reg_entity_child, RegisteredMetric):
+            raise TypeError("reg_entity_child should either be a Registered Device or Metric")
+
+        reg_entity_child.parent = reg_entity_parent
+
+    def _get_entity_hierarchy(self, reg_entity):
+        """
+        :param reg_entity: RegisteredMetric Object
+        :return: Entity hierarchy delimited by '->'
+        """
+        if not isinstance(reg_entity, RegisteredEntity):
+            raise TypeError("RegisteredEntity is expected")
+
+        def extract_hierarchy(reg_entity):
+            """
+            Recursive function to get entity_names
+            :param reg_entity: RegisteredEntity Object
+            :return:
+            """
+            if reg_entity is None:
+                return ""
+            else:
+                return extract_hierarchy(reg_entity.parent) + reg_entity.ref_entity.name + '->'
+        return extract_hierarchy(reg_entity.parent) + reg_entity.ref_entity.name
+
+    def _format_data(self, reg_metric):
+        """
+        :param reg_metric: Registered Metric Object
+        :return: Payload in JSON format
+        """
+        _list = []
+        met_cnt = reg_metric.values.qsize()
+        if met_cnt > 0:
```

Comment:
met_cnt is zero then we`ll be sending a blank metric_data payload? Will it get filtered out later?